unittests/test_schedulers.py

```diff
@@ -544,116 +579,257 @@ def setUp(self):
             stdout=os.path.join(self.workdir, 'testjob.out'),
             stderr=os.path.join(self.workdir, 'testjob.err')
         )
-        # monkey patch `_get_reservation_nodes` to simulate extraction of
+        # monkey patch `_show_nodes` to simulate extraction of
         # slurm nodes through the use of `scontrol show`
-        self.testjob._get_reservation_nodes = self.create_dummy_nodes
+        self.testjob._show_nodes = self.create_dummy_nodes
+        self.testjob._flex_alloc_tasks = 'all'
         self.testjob._num_tasks_per_node = 4
         self.testjob._num_tasks = 0
 
     def tearDown(self):
         os_ext.rmtree(self.workdir)
 
-    def test_valid_constraint(self, expected_num_tasks=8):
-        self.testjob._sched_reservation = 'Foo'
+    def test_not_valid_flex_alloc_tasks(self):
+        self.testjob._flex_alloc_tasks = 'invalid'
+        self.testjob._sched_access = ['--constraint=f1']
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_positive_flex_alloc_tasks(self):
+        self.testjob._flex_alloc_tasks = '12'
+        self.testjob._sched_access = ['--constraint=f1']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 48)
+
+    def test_zero_flex_alloc_tasks(self):
+        self.testjob._flex_alloc_tasks = '0'
+        self.testjob._sched_access = ['--constraint=f1']
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_negative_flex_alloc_tasks(self):
+        self.testjob._flex_alloc_tasks = '-4'
+        self.testjob._sched_access = ['--constraint=f1']
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_sched_access_idle(self):
+        self.testjob._flex_alloc_tasks = 'idle'
+        self.testjob._sched_access = ['--constraint=f1']
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
+
+    def test_constraint_idle(self):
+        self.testjob._flex_alloc_tasks = 'idle'
+        self.testjob.options = ['--constraint=f1']
```

Comment:
The following, I don't think it will work, although valid:

```suggestion
        self.testjob.options = ['--constraint', f1']
```

Reply:
Applied the suggested change